zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -486,65 +523,98 @@ benchmarks! {
+        let (_, market_id) = setup_reported_categorical_market_with_pool::<T>(categories.into())?;
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
+        let market = T::MarketCommons::market(&market_id)?;
     }: {
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_categorical_disputed {
         // d = num. disputes
         let d in 0..T::MaxDisputes::get();
 
         let categories = T::MaxCategories::get();
-        let (caller, market_id) =
-            setup_resolve_common_categorical_after_dispute::<T>(0, 0, categories)?;
+        let (caller, market_id) = setup_reported_categorical_market_with_pool::<T>(categories.into())?;
+        T::MarketCommons::mutate_market(&market_id, |market| {
+            let admin = account("admin", 0, 0);
+            market.dispute_mechanism = MarketDisputeMechanism::Authorized(admin);
+            Ok(())
+        })?;
 
         let categories : u32 = categories.saturated_into();
-        for i in 0..categories.min(d) {
+        for i in 0..d {
             let origin = caller.clone();
-            let disputes = crate::Disputes::<T>::get(market_id);
-            let market = T::MarketCommons::market(&Default::default()).unwrap();
-            T::SimpleDisputes::on_dispute(&disputes, &market_id, &market)?;
+            Pallet::<T>::dispute(
+                RawOrigin::Signed(origin).into(),
+                market_id,
+                OutcomeReport::Categorical((i % 2).saturated_into::<u16>()),
+            )?;
         }
+        let market = T::MarketCommons::market(&market_id)?;
     }: {
+        Pallet::<T>::on_resolution(&market_id, &market)?;
+    } verify {
         let market = T::MarketCommons::market(&market_id)?;
-        let disputes = crate::Disputes::<T>::get(market_id);
-        T::SimpleDisputes::on_resolution(&disputes, &market_id, &market)?
+        assert_eq!(market.status, MarketStatus::Resolved);
     }
 
     internal_resolve_scalar_reported {
         let total_accounts = 10u32;
         let asset_accounts = 10u32;
-        let (_, market_id) =
-            setup_resolve_common_scalar_after_dispute::<T>(total_accounts, asset_accounts)?;
+        let (caller, market_id) = setup_reported_scalar_market_with_pool::<T>()?;
+        generate_accounts_with_assets::<T>(
+            total_accounts,
+            asset_accounts,
+            Asset::ScalarOutcome(market_id, ScalarPosition::Long),
+        )?;
     }: {
         let market = T::MarketCommons::market(&market_id)?;
```

Comment:
I think you want this to be in the scope above, otherwise you're benchmarking the `market` call, as well.